src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2727,7 +2727,7 @@ object Types {
 
         case tp @ AnnotatedType(annot, underlying) =>
           val underlying1 = this(underlying)
-          if (underlying1 eq underlying) tp else underlying1
+          if (underlying1 eq underlying) tp else tp.derivedAnnotatedType(annot, underlying1)
```

Comment:
Ah, yes we do.

Added commit ac61c26 that additionally maps over any types occurring within the annotation tree.